### PR TITLE
lambda.part2.exercise1 (Seliukov)

### PR DESCRIPTION
**Pull request by Duelist256**

### CODE REVIEW

File: src/test/java/lambda/part2/exercise/Exercise1.java [hunk 1]
```diff
@@ -31,33 +31,42 @@ public void sameAgesCheckerUsingBiPredicate() {
...
+         assertTrue(sameAgesChecker.test(person1, person2));
+         assertFalse(sameAgesChecker.test(person1, person3));
+         assertFalse(sameAgesChecker.test(person2, person3));
     }
 
     // TODO метод getFullName: Person -> String, извлекающий из объекта Person строку в формате "имя фамилия".
-    // private static ... getFullName(...) {
+     private static String getFullName(Person person) {
+        return person.getFirstName() + person.getLastName();
+     }
```
Comment (a-filipp0v) [inline src/test/java/lambda/part2/exercise/Exercise1.java]: Person class has it's own getFullName() method.

File: src/test/java/lambda/part2/exercise/Exercise1.java [hunk 2]
```diff
@@ -31,33 +31,42 @@ public void sameAgesCheckerUsingBiPredicate() {
...
+     private static BiFunction<Person, Person, Integer>
+     createExtractorAgeOfPersonWithTheLongestFullName(Function<Person, String> extractor) {
+         return (person1, person2) -> {
+             int person1Length = extractor.apply(person1).length();
+             int person2Length = extractor.apply(person2).length();
+
+             if (person1Length >= person2Length) return person1.getAge();
+             else return person2.getAge();
+         };
+     }
```
Comment (a-filipp0v) [inline src/test/java/lambda/part2/exercise/Exercise1.java]: Can be shorter without separate int variables + with ternary operator.

File: src/test/java/lambda/part2/exercise/Exercise1.java [hunk 3]
```diff
@@ -31,33 +31,42 @@ public void sameAgesCheckerUsingBiPredicate() {
...
 
-        // TODO воспользоваться ссылкой на метод getFullName
-        Function<Person, String> getFullName = null;
+        Function<Person, String> getFullName = Exercise1::getFullName;
 
-        // (Person, Person) -> Integer
-        // TODO воспользоваться методом createExtractorAgeOfPersonWithTheLongestFullName
-        BiFunction<Person, Person, Integer> extractorAgeOfPersonWithTheLongestFullName = null;
+        BiFunction<Person, Person, Integer> extractorAgeOfPersonWithTheLongestFullName =
+                createExtractorAgeOfPersonWithTheLongestFullName(Person::getFullName);
```
Comment (a-filipp0v) [inline src/test/java/lambda/part2/exercise/Exercise1.java]: I guess getFullName variable from line 66 should be used here as a parameter.

File: src/test/java/lambda/part2/exercise/Exercise1.java (hunk 2 above)
Comment (Duelist256) [reply src/test/java/lambda/part2/exercise/Exercise1.java]: I changed if/else to ternary operator. 
But after reading Clean Code I think that using variables is more readable in this case.

**Pull request merged**